src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -598,6 +604,67 @@ public ImmutableArray<byte> GetChecksum()
             return _lazyChecksum;
         }
 
+        /// <summary>
+        /// Produces a hash of this <see cref="SourceText"/> based solely on the contents it contains.  Two different
+        /// <see cref="SourceText"/> instances that are <see cref="ContentEquals"/> will have the same content hash. Two
+        /// instances of <see cref="SourceText"/> with different content are virtually certain to not have the same
+        /// hash.  This hash can be used for fingerprinting of text instances, but does not provide cryptographic
+        /// guarantees.
+        /// </summary>
+        /// <remarks>
+        /// The implementation of this function uses the XxHash128 <see
+        /// href="https://learn.microsoft.com/en-us/dotnet/api/system.io.hashing.xxhash128"/>.  However, this is not
+        /// guaranteed, and future versions of Roslyn may change this.
```

Comment:
I wouldn't put this in the remarks. I think the following paragraph is all we need.

Reply:
sgtm

Reply:
revised. t hanks.